packages/pipeline/src/data_sources/ohlcv_external/crypto_compare.ts

```diff
@@ -0,0 +1,88 @@
+// tslint:disable:no-duplicate-imports
+import { AxiosResponse } from 'axios';
+import axios from 'axios';
+import promiseLimit = require('p-limit');
+
+import { merge, unfold } from 'ramda';
+
+import { TradingPair } from '../../utils/get_ohlcv_trading_pairs';
+
+export interface CryptoCompareOHLCVResponse {
+    Data: CryptoCompareOHLCVRecord[];
+}
+
+export interface CryptoCompareOHLCVRecord {
+    time: number; // in seconds, not milliseconds
+    close: number;
+    high: number;
+    low: number;
+    open: number;
+    volumefrom: number;
+    volumeto: number;
+}
+
+export interface CryptoCompareOHLCVParams {
+    fsym: string;
+    tsym: string;
+    e?: string;
+    aggregate?: string;
+    aggregatePredictableTimePeriods?: boolean;
+    limit?: number;
+    toTs?: number;
+}
+
+// tslint:disable:custom-no-magic-numbers
+const ONE_WEEK = 7 * 24 * 60 * 60 * 1000;
+const ONE_SECOND = 1000;
+
+const HISTORICAL_HOURLY_OHLCV_URI = 'https://min-api.cryptocompare.com/data/histohour';
+
+export class CryptoCompareOHLCVSource {
+    public readonly interval = ONE_WEEK; // the hourly API returns data for one week at a time
+    public readonly default_exchange = 'CCCAGG';
+    private readonly _url: string = HISTORICAL_HOURLY_OHLCV_URI;
+    private readonly _plimit: (
+        fetchFn: () => Promise<AxiosResponse<CryptoCompareOHLCVResponse>>,
+    ) => Promise<AxiosResponse<CryptoCompareOHLCVResponse>>;
+
+    constructor(maxConcurrentRequests: number = 50) {
+        this._plimit = promiseLimit(maxConcurrentRequests);
+    }
+
+    // gets OHLCV records starting from pair.latest
+    public async getAsync(pair: TradingPair): Promise<CryptoCompareOHLCVRecord[]> {
+        const params = {
+            e: this.default_exchange,
+            fsym: pair.fromSymbol,
+            tsym: pair.toSymbol,
+            toTs: Math.floor((pair.latest + this.interval) / ONE_SECOND), // CryptoCompare uses timestamp in seconds. not ms
+        };
+
+        const fetchPromise = this._plimit(() => {
+            // tslint:disable:no-console
+            console.log(`Scraping Crypto Compare with ${JSON.stringify(params)}`);
+            return axios.get<CryptoCompareOHLCVResponse>(this._url, { params });
+        });
+
+        const resp = await Promise.resolve(fetchPromise);
+
+        if (Object.keys(resp.data.Data).length < 2) {
+            console.log(`No data found for ${JSON.stringify(params)}`);
```

Comment:
I disabled no-console - didn't know about logUtils! Thank you!

Reply:
I would argue that `logUtils.log` should not exist. Let me explain.

Our goal with the `no-console` TSLint rule is to disallow logging in code where it does not belong. The existence of `logUtils.log` is counteractive to this goal because it *implicitly* bypasses the linter rule. In other words, if you call `logUtils.log` in your code, the linter will not warn you. As we add more and more developers on the team, someone is bound to call `logUtils.log` in a place where it really does not belong and they will have no idea that it is against our conventions. They will probably just think that `logUtils.log` is what we use for logging.

Here's what I propose instead:

Just use `console.log`. Normally the linter will warn you, but you can explicitly disable it if you feel that you are working on a part of the codebase where logging is appropriate. That *explicit* step of adding a `// tslint:disable` comment is important because it makes it less likely that someone will do this by accident.

@fabioberger what do you think?

Reply:
Yeah, I'd be fine with that, although I feel like the argument is similar. The reason we have `logUtils.log` is because most dev's use `console.log` for debugging and might accidentally leave it in the code. `logUtils.log` is not "front-of-mind" and therefore less likely to be left accidentally (although still a possibility). I do think that adding the `// tslint:disable-next-line` adds even more friction so it could be even better at achieving our stated aim. Let's bring this up at the dev meeting next week, and if the majority agrees, let's replace `logUtils.log` with `console.log` & tslint disable-next-line.